catchemall/src/App.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import './App.css';
+import {BrowserRouter as Router, Route, Switch, Link} from 'react-router-dom';
+
+let pokeNumber = 1;
+
+const pokeTopTen = function () {
+  for (let i = 0; i < 10; i++) {
+    pokeNumber++;
+  }
+ } 
+
+const pokeNameGetter = function(poke) { 
+  return poke.forms[0].name;
+}
+
+
+class Pokemon extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {},
+      isLoading: true,
+      hasError: false
+    }
+  }
+  componentDidMount() {
+    fetch(`https://pokeapi.co/api/v2/pokemon/${pokeNumber}`)
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        this.setState({
+          data: data,
+          isLoading: false
+        });
+      })
+      .catch(error => {
+        this.setState({
+          hasError: true,
+          isLoading: false
+        });
+      });
+  }
+
+  render() {
+    if (this.state.isLoading) {
+      return <div>Loading...</div>;
+    }
+
+    if (this.state.hasError) {
+      return <div>ERROR, please reload and try again</div>;
+    }
+
+    const names = Object.keys(this.state.data)
+      .map((names, idx) => <li key={idx}>{names}</li>);
+
```

Comment:
Then inside of a new component pokeDetails for example you could grab the data for that specific pokemon using the following fetch: 
`fetch(`https://pokeapi.co/api/v2/pokemon/${this.props.match.params.name}`)`